ansible_base/authentication/utils/authentication.py

```diff
@@ -20,6 +21,51 @@ def setting(self, *args, **kwargs):
         return ["username", "email"]
 
 
+def migrate_from_existing_authenticator(uid, alt_uid, authenticator: Authenticator, preferred_username=None):
+    uid_filter = [
+        uid,
+    ]
+    if alt_uid:
+        uid_filter = [uid, alt_uid]
+
+    if authenticator.auto_migrate_users_from.filter(type__endswith="saml").exists():
+        # SAML puts prepends all the UIDs with IdP. Adding this to the search criteria will
+        # allow us to find SAML accounts that match the UID.
+        uid_filter.append("IdP:" + uid)
```

Comment:
Would it hurt to just add this with out the exists filter to save a query?

Reply:
No problem. That's how I had it originally.